R/check_dl_file.R

```diff
@@ -0,0 +1,60 @@
+#' @title Verify and download data files
+#' 
+#' @description 
+#' Checks if local data files as defined in master file exist and match MD5 hash. Downloads data if necessary.
+#' 
+#' @param master_file Character path to master file
+#' @param fname Character vector of specific file names to check
+#' @param md5check boolean
+#' @param dest Character path to download destination
+#' 
+#' @import httr
+#' @import tools
+#' 
+#' @export
+
+check_dl_file = function(master_file, fname = NULL, md5check = TRUE, dest=local_path()){
+  files = read.csv(master_file)
+  if(!is.null(fname)){
+   files = files[files$filename == fname,]
+  }
+  apply(files, 1, function(x){
+    if(!file.exists(file.path(dest, x["filename"]))){
+      r = RETRY("GET", url=x["URL"], write_disk(file.path(dest,x["filename"]), overwrite=TRUE), times = 2)
+      stop_for_status(r)
+      if(md5sum(file.path(dest, x["filename"])) != x["MD5"]){
+        file.remove(file.path(dest, x["filename"]))
+        stop("Hash mismatch error when downloading data - please try again later or submit a bug report.")
+      }
+    }
+  })
+  if(md5check){
+    apply(files, 1, function(x){
+     if(md5sum(file.path(dest, x["filename"])) != x["MD5"]){
+       r = RETRY("GET", url=x["URL"], write_disk(file.path(dest, x["filename"]), overwrite=TRUE))
+       stop_for_status(r)
+       if(md5sum(file.path(dest, x["filename"])) != x["MD5"]){
+         file.remove(file.path(dest, x["filename"]))
```

Comment:
We may not get here if it stops at `stop_for_status`. Is that ok? Or do we want to move or eliminate stop_for_status? Could just fully rely on the MD5 hash (and existence of the file?) as an indication if the request was successful.